src/Extensions/Compatability/Validation/Code.cs

```diff
@@ -0,0 +1,515 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Omex.Extensions.Compatability.Validation
+{
+	/// <summary>
+	/// Code validation
+	/// </summary>
+	public static class Code
+	{
+		/// <summary>
+		/// Check object's state, throws specified Exception type if state evaluates to false.
+		/// </summary>
+		/// <typeparam name="TException">Exception to throw</typeparam>
+		/// <param name="state">Object's state</param>
+		/// <param name="errorMessage">Error message</param>
+		/// <param name="tagId">Tag Id to log, leave null if no logging is needed</param>
```

Comment:
Do we really want to support this scenario? Why would you not want to log. I'm concerned that this can just result in errors that aren't expected

Reply:
I don't want to :) but this is `Compatibility` package added only for the transition period so I've decided it should support the same behavior that was before.